src/Native/Windows/CMakeLists.txt

```diff
@@ -1,23 +1,123 @@
 
-if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 3.5)
-    add_compile_options(-Wno-unreachable-code)
+# The following options are set by the razzle build
+add_compile_options(/d2Zi+)       # make optimized builds debugging easier
+add_compile_options(/nologo)      # Suppress Startup Banner
+add_compile_options(/Oi)          # enable intrinsics
+add_compile_options(/Oy-)         # disable suppressing of the creation of frame pointers on the call stack for quicker function calls
+add_compile_options(/U_MT)        # undefine the predefined _MT macro
+add_compile_options(/GF)          # enable read-only string pooling
+add_compile_options(/Gm-)         # disable minimal rebuild
+add_compile_options(/EHa)         # enable C++ EH (w/ SEH exceptions)
+add_compile_options(/Zp8)         # pack structs on 8-byte boundary
+add_compile_options(/Gy)          # separate functions for linker
+add_compile_options(/Zc:wchar_t-) # C++ language conformance: wchar_t is NOT the native type, but a typedef
+add_compile_options(/Zc:forScope) # C++ language conformance: enforce Standard C++ for scoping rules
+add_compile_options(/GR-)         # disable C++ RTTI
+add_compile_options(/FC)          # use full pathnames in diagnostics
+add_compile_options(/MP)          # Build with Multiple Processes (number of processes equal to the number of processors)
+add_compile_options(/GS)          # Buffer Security Check
+add_compile_options(/Zm200)       # Specify Precompiled Header Memory Allocation Limit of 150MB
+add_compile_options(/Zi)          # enable debugging information
+add_compile_options(/Zl)          # enable debugging information
+add_compile_options(/wd4960 /wd4961 /wd4603 /wd4627 /wd4838 /wd4456 /wd4457 /wd4458 /wd4459 /wd4091 /we4640)
+
+if (CMAKE_SYSTEM_PROCESSOR STREQUAL "x86")
+    add_compile_options(/Gz)
+endif ()
+if(NOT "${__BuildArch}" STREQUAL "arm64")
+    # enable control-flow-guard support for native components for non-Arm64 builds
+    add_compile_options(/guard:cf) 
+    set(__LinkArgs "${__LinkArgs} /guard:cf")
 endif ()
 
-if (CMAKE_SYSTEM_PROCESSOR STREQUAL x86_64 OR CMAKE_SYSTEM_PROCESSOR STREQUAL amd64)
+SET (CMAKE_C_STANDARD_LIBRARIES "") # do not link against standard win32 libs i.e. kernel32, uuid, user32, etc.
+SET (CMAKE_CXX_STANDARD_LIBRARIES "") # do not link against standard win32 libs i.e. kernel32, uuid, user32, etc.
+
+# Statically linked CRT (libcmt[d].lib, libvcruntime[d].lib and libucrt[d].lib) by default. This is done to avoid
+# linking in VCRUNTIME140.DLL for a simplified xcopy experience by reducing the dependency on VC REDIST.
+#
+# For Release builds, we shall dynamically link into uCRT [ucrtbase.dll] (which is pushed down as a Windows Update on downlevel OS) but
+# wont do the same for debug/checked builds since ucrtbased.dll is not redistributable and Debug/Checked builds are not
+# production-time scenarios.
+if (UPPERCASE_CMAKE_BUILD_TYPE STREQUAL "RELEASE")
```

Comment:
I see there are other CMake Release build types (RelWithDebInfo and MinSizeRel).  We just don't use them and hence we are OK with this check?